Content.Client/Botany/PlantAnalyzer/PlantAnalyzerWindow.xaml.cs

```diff
@@ -0,0 +1,210 @@
+
+
+        // Section 3.5: They are putting chemicals in the water!
+        if (msg.TrayData?.Chemicals != null)
+        {
+            var count = msg.TrayData.Chemicals.Count;
+            var holder = ContainerLabel.Text;
+            var chemicals = PlantAnalyzerLocalizationHelper.ChemicalsToLocalizedStrings(msg.TrayData.Chemicals, _prototypeManager);
+            if (count == 0)
+                ChemicalsInWaterLabel.Text = Loc.GetString("plant-analyzer-soil-empty", ("holder", holder));
+            else
+                ChemicalsInWaterLabel.Text = Loc.GetString("plant-analyzer-soil", ("count", count), ("holder", holder), ("chemicals", chemicals));
+
+            ChemicalsInWaterBox.Visible = true;
+        }
+        else
+        {
+            ChemicalsInWaterBox.Visible = false;
+        }
+        ChemicalsInWaterDivider.Visible = ChemicalsInWaterBox.Visible;
+
+        // Section 4: Tolerances part 2.
+        if (msg.TolerancesData is not null)
+        {
+            (string, string)[] parameters = [
+                ("seedName", SeedLabel.Text),
+                ("gases", PlantAnalyzerLocalizationHelper.GasesToLocalizedStrings(msg.TolerancesData.ConsumeGasses, _prototypeManager)),
+                ("kpa", msg.TolerancesData.IdealPressure.ToString("0.00")),
+                ("kpaTolerance", msg.TolerancesData.PressureTolerance.ToString("0.00")),
+                ("temp", msg.TolerancesData.IdealHeat.ToString("0.00")),
+                ("tempTolerance", msg.TolerancesData.HeatTolerance.ToString("0.00")),
+                ("lightLevel", msg.TolerancesData.IdealLight.ToString("0.00")),
+                ("lightTolerance", msg.TolerancesData.LightTolerance.ToString("0.00"))
+            ];
+            EnvironmentLabel.Text = msg.TolerancesData.ConsumeGasses.Count == 0
+                ? msg.TolerancesData.IdealHeat - msg.TolerancesData.HeatTolerance <= 0f && msg.TolerancesData.IdealPressure - msg.TolerancesData.PressureTolerance <= 0f
+                    ? Loc.GetString("plant-analyzer-component-environemt-void", [.. parameters])
+                    : Loc.GetString("plant-analyzer-component-environemt", [.. parameters])
+                : Loc.GetString("plant-analyzer-component-environemt-gas", [.. parameters]);
+
+            EnvironmentBox.Visible = true;
+        }
+        else
+        {
+            EnvironmentBox.Visible = false;
+        }
+        EnvironmentDivider.Visible = EnvironmentBox.Visible;
+
+        // Section 5: Output
+        if (msg.ProduceData is not null)
+        {
+            var gases = PlantAnalyzerLocalizationHelper.GasesToLocalizedStrings(msg.ProduceData.ExudeGasses, _prototypeManager);
+            var (produce, producePlural) = PlantAnalyzerLocalizationHelper.ProduceToLocalizedStrings(msg.ProduceData.Produce, _prototypeManager);
+            var chemicals = PlantAnalyzerLocalizationHelper.ChemicalsToLocalizedStrings(msg.ProduceData.Chemicals, _prototypeManager);
```

Comment:
this should put colour tags around the individual items instead of the list as a whole going into a colour tag

Reply:
Have you looked at the FTL? It's already complex enough as is. Having the "," and "and/or" colored in a list is not an issue in my opinion.

Reply:
fair enough